compiler/src/lib.rs

```diff
@@ -22,6 +25,10 @@ pub fn no_callback<T>() -> Option<fn(&str) -> Option<T>> {
     None
 }
 
+type Commits<'a, T> = Vec<(&'a str, Vec<T>)>;
```

Comment:
This is used elsewhere without the alias, and Proof is defined in the `backend` crate as well. Can we put this somewhere to be reused? Probably in `backend` for `Proof` and in  `executor` or `number` (not ideal but that's where serialization is now) for the columns?

Reply:
I kept these private for now just to use them locally, as I'm not sure this is the best solution for the entire code base. Should I still do it?